File: app/src/main/java/xyz/volgoak/wordlearning/recycler/SingleChoiceMode.java
package xyz.volgoak.wordlearning.recycler;

import android.os.Bundle;

import java.util.List;

/**
 * Created by Volgoak on 20.08.2017.
 */

public class SingleChoiceMode implements ChoiceMode {

    private int checkedPosition = -1;

    @Override
    public void setChecked(int position, boolean checked) {
        if(checked){
            checkedPosition = position;
        }
    }

    @Override
    public boolean isChecked(int position) {
        return position == checkedPosition;
    }

    @Override
    public void onSaveInstanceState(Bundle instanceState) {

    }

    @Override
    public void restoreInstanceState(Bundle instanceState) {

    }

    @Override
    public int getCheckedCount() {
        return checkedPosition == -1 ? 0 : 1;
    }

    @Override
    public void clearChecks() {
        checkedPosition = -1;
    }

    @Override
    public List<Integer> getCheckedList() {
        return null;
    }

    @Override
    public int getCheckedPosition() {
        return checkedPosition;
    }
}

File: app/src/main/java/xyz/volgoak/wordlearning/adapter/choice_mode/SingleChoiceMode.kt
package xyz.volgoak.wordlearning.adapter.choice_mode

import android.os.Bundle

/**
 * Created by Volgoak on 20.08.2017.
 */

class SingleChoiceMode : ChoiceMode {

    override var checkedPosition = -1

    override val checkedCount: Int
        get() = if (checkedPosition == -1) 0 else 1

    override val checkedList: List<Int> = listOf()

    override fun setChecked(position: Int, checked: Boolean) {
        if (checked) {
            checkedPosition = position
        }
    }

    override fun isChecked(position: Int): Boolean {
        return position == checkedPosition
    }

    override fun onSaveInstanceState(instanceState: Bundle) {}

    override fun restoreInstanceState(instanceState: Bundle) {}

    override fun clearChecks() {
        checkedPosition = -1
    }
}
